site/app/controllers/PollController.php

```diff
@@ -398,4 +399,64 @@ public function viewResults($poll_id) {
             )
         );
     }
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/polls/getPollExport", methods={"GET"})
+     * @AccessControl(role="INSTRUCTOR")
+     * @return JsonResponse
+     */
+    public function getPollExportData(): JsonResponse {
+        $polls = $this->core->getQueries()->getPolls();
+        return JsonResponse::getSuccessResponse(PollUtils::getPollExportData($polls));
+    }
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/polls/importPolls", methods={"POST"})
+     * @AccessControl(role="INSTRUCTOR")
+     * @return RedirectResponse
+     */
+    public function importPollsFromJSON(): RedirectResponse {
+        $filename = $_FILES["polls_file"]["tmp_name"];
+        $file = fopen($filename, "r");
+        $contents = fread($file, filesize($filename));
+        $polls = json_decode($contents, true);
+        if (json_last_error() != JSON_ERROR_NONE) {
+            $this->core->addErrorMessage("Failed to read file. Make sure the file is the right format");
+            return new RedirectResponse($this->core->buildCourseUrl(['polls']));
+        }
+        $num_imported = 0;
+        $num_errors = 0;
+        foreach ($polls as $poll) {
+            if (
+                !array_key_exists("name", $poll) || !array_key_exists("question", $poll)
+                || !array_key_exists("responses", $poll) || !array_key_exists("correct_responses", $poll)
+                || !array_key_exists("release_date", $poll)
```

Comment:
Put all elements on their own lines. This will (eventually) be a lint rule:

```suggestion
                !array_key_exists("name", $poll)
                || !array_key_exists("question", $poll)
                || !array_key_exists("responses", $poll)
                || !array_key_exists("correct_responses", $poll)
                || !array_key_exists("release_date", $poll)
```